src/Microsoft.DotNet.Darc/src/Darc/Operations/CloneOperation.cs

```diff
@@ -0,0 +1,204 @@
+using CSharpx;
+using Microsoft.DotNet.Darc.Helpers;
+using Microsoft.DotNet.Darc.Options;
+using Microsoft.DotNet.DarcLib;
+using Microsoft.DotNet.DarcLib.Helpers;
+using Microsoft.Extensions.Logging;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Microsoft.DotNet.Darc.Operations
+{
+    internal class CloneOperation : Operation
+    {
+        CloneCommandLineOptions _options;
+
+        public CloneOperation(CloneCommandLineOptions options) : base(options)
+        {
+            _options = options;
+        }
+
+        public override async Task<int> ExecuteAsync()
+        {
+            try
+            {
+                EnsureOptionsCompatibility(_options);
+                RemoteFactory remoteFactory = new RemoteFactory(_options);
+                HashSet<StrippedDependency> seenDependencies = new HashSet<StrippedDependency>();
```

Comment:
I don't think you actually need this.  Your cloning code skips a directory if it already exists, so shouldn't it do the "already been cloned" filtering by default?

Reply:
Yeah, this was leftover from look at dependencies in repos after cloning them.  Removed.